server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -697,12 +667,6 @@ private void endSnapshot(SnapshotsInProgress.Entry entry, Metadata metadata) {
             return;
         }
         final Snapshot snapshot = entry.snapshot();
-        if (entry.repositoryStateId() == RepositoryData.UNKNOWN_REPO_GEN) {
```

Comment:
This is unnecessary now, we will not pass a snapshot with `UNKNOWN_REPO_GEN` to this method any longer, we just remove it from the cluster state in `processWaitingShardsAndRemovedNodes` and are done with it.